web/src/app/services/selection.service.ts

```diff
@@ -1,20 +1,24 @@
 import { Injectable } from '@angular/core';
 
 import 'rxjs/Rx';
-import {Subject,Subscription, Subscriber} from 'rxjs/Rx';
+import {Subject, Subscription, Subscriber} from 'rxjs/Rx';
 
-import { Section } from 'yacs-api-client';
-import { Listing } from 'yacs-api-client';
+import { Section, Listing, Term } from 'yacs-api-client';
 import { SidebarService } from './sidebar.service';
-
+import { SelectedTermService } from './selected-term.service';
 
 @Injectable()
 export class SelectionService {
 
   private clickEvent = new Subject();
 
    constructor (
-    public sidebarService : SidebarService) { }
+    public sidebarService: SidebarService,
+    protected selectedTermService: SelectedTermService) {
+    this.selectedTermService.subscribeToActiveTerm((term: Term) => {
```

Comment:
Will this clear the selections when the page loads and the active term is initially set?

Reply:
It will not; if the active term is not set in local storage on load, the service sets it internally without posting an update through the observable. Additionally, the active term is only stored in local storage if it differs from the latest term.

Reply:
Awesome